src/main/java/io/github/hapjava/server/impl/HomekitRoot.java

```diff
@@ -83,11 +85,19 @@ public class HomekitRoot {
     this(
         label, DEFAULT_ACCESSORY_CATEGORY, webHandler, authInfo, new JmdnsHomekitAdvertiser(jmdns));
   }
+
+  public synchronized void batchUpdate() {
+    if (this.nestedBatches == 0) madeChanges = false;
+    ++this.nestedBatches;
+  }
+
+  public synchronized void completeUpdateBatch() {
```

Comment:
So, nestedBatches can be negative number if completeUpdate called more often than updateBatch

Reply:
Theoretically. That'd probably be a violation of the interface to not have matching batchUpdate/completeUpdateBatch calls.